### ISSUE TITLE
[FR] - Dynamic JS Event Values within form UI

### ISSUE THREAD

**Issue by ljclark** [verbb/formie#888]
## What are you trying to do?
It'd be great to be able to use the values of your form fields in the JS Events utility within the forms UI, so you can dynamically populate the value of your dataLayer events. 

## What's your proposed solution?
Ideally, this could be as simple as referencing the form handle value in the Value column. For more complex field types, perhaps  approaching it similar to how you'd build out a dynamic slug within the core Craft UI for new sections, using Twig.

**Comment by engram-design** [comment]
Yep, this might be tricky to implement, but it's also a common thing. 

The main reason for this is that it all needs to be done client-side with JS, and fetching values needs to be done there also, particularly if the form uses Ajax. For some fields, there's the reliance on server-side normalization. We'll need a foolproof method of referring to a field by handle (that can't be overridden by template overrides), and also deal with the many different types of inputs.

**Comment by engram-design** [comment]
Added in [4.0.0-beta.8](https://github.com/verbb/formie/releases/tag/4.0.0-beta.8). We've also revamped the client events UI with presets.

<img width="783" height="790" alt="Image" src="https://github.com/user-attachments/assets/4bb8b844-201f-4d3e-9a3a-933477abe11d" />

<img width="1104" height="562" alt="Image" src="https://github.com/user-attachments/assets/33871640-a58c-45b3-bbc9-0de86d72bf88" />
